### PR TITLE
Mockery deprecation

### PR DESCRIPTION
**Pull request by wxing1292**
* Use goimports instead of gofmt for better imports management
* Deprecate domain cache mockery

### CODE REVIEW

File: service/history/nDCStateRebuilder_test.go [hunk 1]
```diff
@@ -308,7 +307,7 @@ func (s *nDCStateRebuilderSuite) TestRebuild() {
 		Size:          historySize2,
 	}, nil).Once()
 
-	s.mockDomainCache.On("GetDomainByID", targetDomainID).Return(cache.NewGlobalDomainCacheEntryForTest(
+	s.mockDomainCache.EXPECT().GetDomainByID(targetDomainID).Return(cache.NewGlobalDomainCacheEntryForTest(
```
Comment (yycptt) [inline service/history/nDCStateRebuilder_test.go]: `AnyTimes()`

File: service/history/queryRegistry_test.go [hunk 2]
```diff
@@ -23,9 +23,10 @@
 package history
 
 import (
-	"github.com/uber/cadence/common"
 	"testing"
 
+	"github.com/uber/cadence/common"
+
```
Comment (yycptt) [inline service/history/queryRegistry_test.go]: remove

File: service/history/stateBuilder_test.go [hunk 3]
```diff
@@ -498,7 +495,7 @@ func (s *stateBuilderSuite) TestApplyEvents_EventTypeWorkflowExecutionContinuedA
 	s.mockMutableState.EXPECT().ClearStickyness().Times(1)
 
 	// new workflow domain
-	s.mockDomainCache.On("GetDomain", testParentDomainName).Return(testGlobalParentDomainEntry, nil).Once()
+	s.mockDomainCache.EXPECT().GetDomain(testParentDomainName).Return(testGlobalParentDomainEntry, nil).AnyTimes()
```
Comment (yycptt) [inline service/history/stateBuilder_test.go]: `Times(1)`?

File: service/history/timerQueueProcessor2_test.go [hunk 4]
```diff
@@ -116,13 +117,10 @@ func (s *timerQueueProcessor2Suite) SetupTest() {
 	s.mockShardManager = &mocks.ShardManager{}
 	s.mockHistoryV2Mgr = &mocks.HistoryV2Manager{}
 	s.mockVisibilityMgr = &mocks.VisibilityManager{}
-	s.mockDomainCache = &cache.DomainCacheMock{}
+
```
Comment (yycptt) [inline service/history/timerQueueProcessor2_test.go]: remove

File: service/matching/matcher_test.go [hunk 5]
```diff
@@ -28,8 +28,8 @@ import (
 
 	"github.com/golang/mock/gomock"
 	"github.com/pborman/uuid"
-	"github.com/stretchr/testify/mock"
 	"github.com/stretchr/testify/suite"
+
```
Comment (yycptt) [inline service/matching/matcher_test.go]: remove

File: service/matching/matcher_test.go [hunk 6]
```diff
@@ -391,8 +391,8 @@ func (t *MatcherTestSuite) newDomainCache() cache.DomainCache {
 		&persistence.DomainConfig{},
 		"",
 		nil)
-	dc := &cache.DomainCacheMock{}
-	dc.On("GetDomainByID", mock.Anything).Return(entry, nil)
+	dc := cache.NewMockDomainCache(t.controller)
+	dc.EXPECT().GetDomainByID(gomock.Any()).Return(entry, nil)
```
Comment (yycptt) [inline service/matching/matcher_test.go]: AnyTimes()

File: service/matching/matchingEngine_test.go [hunk 7]
```diff
@@ -31,6 +31,7 @@ import (
 	"time"
 
 	"github.com/golang/mock/gomock"
+
```
Comment (yycptt) [inline service/matching/matchingEngine_test.go]: remove. and L55

File: service/matching/taskListManager_test.go [hunk 8]
```diff
@@ -27,8 +27,9 @@ import (
 	"testing"
 	"time"
 
-	"github.com/stretchr/testify/mock"
+	"github.com/golang/mock/gomock"
 	"github.com/stretchr/testify/require"
+
```
Comment (yycptt) [inline service/matching/taskListManager_test.go]: remove.

File: service/worker/replicator/processor_test.go [hunk 9]
```diff
@@ -27,8 +27,10 @@ import (
 
 	"github.com/golang/mock/gomock"
 	"github.com/stretchr/testify/mock"
+	"github.com/stretchr/testify/require"
 	"github.com/stretchr/testify/suite"
 	"github.com/uber-go/tally"
+
```
Comment (yycptt) [inline service/worker/replicator/processor_test.go]: remove.

File: tools/cli/adminElasticSearchCommands.go [hunk 10]
```diff
@@ -25,15 +25,16 @@ import (
...
 	"math"
 	"net/http"
 	"os"
 	"strconv"
 	"strings"
 	"time"
 
+	"github.com/uber/cadence/common/clock"
+	"github.com/uber/cadence/common/tokenbucket"
+
```
Comment (yycptt) [inline tools/cli/adminElasticSearchCommands.go]: remove.

File: common/persistence/cassandra/factory.go [hunk 11]
```diff
@@ -21,9 +21,10 @@
 package cassandra
 
 import (
-	"github.com/uber/cadence/common"
 	"sync"
 
+	"github.com/uber/cadence/common"
+
```
[the remark this answers is not in the document]
Comment (wxing1292) [reply common/persistence/cassandra/factory.go]: the empty line is from goimports, which will be the default formatter

File: common/xdc/historyRereplicator_test.go [hunk 12]
```diff
@@ -53,14 +53,14 @@ type (
...
-		mockDomainCache     *cache.DomainCacheMock
 		mockAdminClient     *adminservicetest.MockClient
 		mockHistoryClient   *historyservicetest.MockClient
 		serializer          persistence.PayloadSerializer
 		logger              log.Logger
 
-		controller   *gomock.Controller
-		rereplicator *HistoryRereplicatorImpl
+		controller      *gomock.Controller
+		mockDomainCache *cache.MockDomainCache
```
[the remark this answers is not in the document]
Comment (wxing1292) [reply common/xdc/historyRereplicator_test.go]: gomocks are grouped together, simpler for later mockery deprecation

File: service/history/stateBuilder_test.go (hunk 3 above)
Comment (wxing1292) [reply service/history/stateBuilder_test.go]: number of getter calls does not matter

File: service/history/timerQueueProcessor2_test.go (hunk 4 above)
Comment (wxing1292) [reply service/history/timerQueueProcessor2_test.go]: i will make change to timer builder in the next PR, the test will be rewritten

**Pull request merged**